### DISCUSSION TITLE
capstone telebot project

### DISCUSSION THREAD

**Discussion by rosintanov** [teamalgoritma/community#61] [Q&A] [answered]
Nama: Rosinta 
Pertanyaan :  heroku tidak bisa dihubungan dengan github

<img width="613" alt="image" src="https://user-images.githubusercontent.com/101467607/164618856-27864d5f-92de-4f68-bfac-0cd42a655105.png">

**Comment by yosiaazarya** [comment]
Apakah Ibu sudah push push ke github terlebih dahulu sebelum di connect dengan heroku? Pastikan github repo yang dihubungkan sudah memiliki file-file yang diperlukan untuk deployment dengna heroku.

**Reply by rosintanov** [reply]
sudah semua

**Reply by yosiaazarya** [reply]
Pada github repositori telebot Ibu, tampaknya belum ada file yang di upload. Mohon untuk di check kembali filenya agar github reponya dapat terhubung ke heroku. Terima kasih
![image](https://user-images.githubusercontent.com/67218587/164648865-16574a77-23b9-4b2c-95fd-6f6e6388a1a2.png)

**Reply by rosintanov** [reply]
sudah pernah di push, tapi tetap hal yg sama terjadi di heroku. 
barusan push ulang tapi jadi seperti ini. cara remove manually nya itu gimana ya pak? 

<img width="457" alt="image" src="https://user-images.githubusercontent.com/101467607/164648975-6ac933ca-b6d3-4b52-a196-17764e73bc44.png">

**Reply by yosiaazarya** [reply]
Ibu dapat membuat repositori baru, kemudian melakukan upload file terlebih dahulu di github, setelah semua file berhasil terupload, maka Ibu bisa connect github repo baru tersebut dengan herokunya ya Ibu.

**Comment by kevbow** [comment]
Halo Bu Rosinta, 

Harap untuk melakukan upload file terlebih dahulu secara manual ke Github, baru setelah itu coba lagi untuk menghubungkannya ke Heroku. Karena repository yang Ibu gunakan untuk dihubungkan ke Heroku masih kosong, maka tidak ada yang bisa diproses oleh Heroku. 
Demikian yang dapat kami sampaikan, terima kasih.

**Comment by rosintanov** [comment]
Pak Kevin dan Pak Yosia, 

sebelumnya saya sudah menyampaikan bahwa file nya sudah pernah di upload, namun tetap tdk bisa autorished heroku. 
upload file pernah dilakukan secara manual ataupun push from vscode. 
namun hasilnya tetap sama. 
<img width="620" alt="image" src="https://user-images.githubusercontent.com/101467607/165222099-835bedc3-ffcd-4f50-8fb1-09b1aca7c225.png">

**Reply by yosiaazarya** [reply]
Dapat diulangi kembali dengan cara membuat github repository baru, kemudian melakukan upload file-file yang diperlukan ke dalam github repository baru tersebut, kemudian menyambungkan ke dalam heroku. Hal ini perlu dilakukan karena error yang Ibu lampirkan terjadi karena github repository yang dicoba untuk connect dengan heroku di indikasi kosong. Terima kasih.

**Comment by yosiaazarya** [comment, marked as answer]
Solusi yang dapat Ibu lakukan untuk deployment ke heroku adalah dengan menggunakan heroku CLI (tidak connect dengan github); berikut adalah langkah-langkah yang dapat Ibu lakukan:

1. Install Heroku CLI (Command Line Interface) disini : https://devcenter.heroku.com/articles/heroku-cli
2. Lakukan login di heroku CLI (dapat dibuka dengan CMD), dengan perintah `heroku login` 
3. Pindahkan directory ke tempat file project berada
4. Tambahkan remote baru dengan perintah `heroku git:remote -a example-app` ; gantilah "example-app" dengan nama aplikasi di heroku (pastikan sudah membuat aplikasi di heroku)
5. Pastikan remote sudah tersedia dengan perintah `git remote -v`
6. Push file ke heroku dengan cara `git push heroku head:master` / `git push heroku head:main`
7. Progress deployment akan terlihat di dashboard heroku

Demikian adalah solusi yang dapat kami berikan. Semoga dengan solusi ini dapat mengatasi permasalahan yang ada. Terima kasih.
